examples/demo.rs

```diff
@@ -254,6 +268,15 @@ impl ApplicationHandler<()> for State {
                 self.latest_profiler_results =
                     profiler.process_finished_frame(queue.get_timestamp_period());
                 console_output(&self.latest_profiler_results, device.features());
+                #[cfg(feature = "puffin")]
+                {
+                    let mut gpu_profiler = PUFFIN_GPU_PROFILER.lock().unwrap();
+                    wgpu_profiler::puffin::output_frame_to_puffin(
+                        &mut gpu_profiler,
+                        self.latest_profiler_results.as_deref().unwrap_or_default(),
+                    );
+                    gpu_profiler.new_frame();
+                }
```

Comment:
It looks like frames come in bursts or there's regular frame drops.
![image](https://github.com/user-attachments/assets/db92b3b2-77b8-4e20-a348-128012648dee)

Makes me wonder if this hints at some issue in wgpu-profiler, but maybe that's just an artifact of the surface mode under this very weak gpu load. Might warrant some experimentation..

Anyways, I think there's something already kinda wrong here to begin with and now very visible: I believe we should process finished frames until None shows up. All of them should be reported to puffin and only the last one be displayed to the console 🤔

Reply:
That's strange, I'm not quite seeing the same thing, but I'm seeing something similar. I've got 2 monitors on my PC, one is 4k and one is 1440p. If I move either the demo window or the profiler window between my monitors, I'll see a couple of blank frames like you've shown in your screenshot. Here's what it looks like:
![image](https://github.com/user-attachments/assets/f26f7071-5dab-4fb8-b340-38410d35ab08)